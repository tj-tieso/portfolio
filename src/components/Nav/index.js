import React, { useEffect } from "react";
import { capitalizeFirstLetter } from "../../utils/helpers";
// import heroImg from "../../assets/cover/heroImg.jpg";

function Nav(props) {
  const { navCategories = [], setNavCategory, currentNavCategory } = props;

  useEffect(() => {
    document.title = capitalizeFirstLetter(currentNavCategory.name);
  }, [currentNavCategory]);

  return (
    <header>
      <a data-testid="link" href="/">
        Tieso
      </a>
      <nav>
        <ul>
          {navCategories.map((navCategory) => (
            <li key={navCategory.name}>
              <a
                className={`${
                  currentNavCategory.name === navCategory.name && "navActive"
                }`}
                onClick={() => {
                  setNavCategory(navCategory);
                }}
                data-testid={navCategory.name}
                href={`# ${navCategory.name}`}
              >
                {capitalizeFirstLetter(navCategory.name)}
              </a>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

export default Nav;
