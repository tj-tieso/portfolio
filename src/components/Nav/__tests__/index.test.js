import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

afterEach(cleanup);

describe("Nav Component : ", () => {
  it("renders", () => {
    render(<Nav />);
  });

  it("matches snapshot DOM node structure", () => {
    // asFragment functionreturns a snapshot of the tested component
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Links are Visible", () => {
  it("inserts text into links", () => {
    const { getByTestId } = render(<Nav />);
    expect(getByTestId("link")).toHaveTextContent("Tieso");
    expect(getByTestId("about")).toHaveTextContent("About");
    expect(getByTestId("portfolio")).toHaveTextContent("Portfolio");
    expect(getByTestId("contact")).toHaveTextContent("Contact");
  });
});
