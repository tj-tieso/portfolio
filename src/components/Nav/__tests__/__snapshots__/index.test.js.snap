// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component :  matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header>
    <a
      data-testid="link"
      href="/"
    >
      Tieso
    </a>
    <nav>
      <ul>
        <li
          class="navActive"
        >
          <a
            data-testid="about"
            href="# about"
          >
            About
          </a>
        </li>
        <li
          class="false"
        >
          <a
            data-testid="portfolio"
            href="# portfolio"
          >
            Portfolio
          </a>
        </li>
        <li
          class="false"
        >
          <a
            data-testid="contact"
            href="# contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
